nntrainer/dataset/func_data_producer.cpp

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   func_data_producer.cpp
+ * @date   12 July 2021
+ * @brief  This file contains various data producers from a callback
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#include <func_data_producer.h>
+
+#include <nntrainer_error.h>
+
+namespace nntrainer {
+
+FuncDataProducer::FuncDataProducer(datagen_cb datagen_cb, void *user_data_) :
+  cb(datagen_cb),
+  user_data(user_data_) {}
+
+FuncDataProducer::~FuncDataProducer() {}
+
+const std::string FuncDataProducer::getType() const {
+  return FuncDataProducer::type;
+}
```

Comment:
Could you list up how many FuncDataProducer Types are?

Reply:
There is one type for now, later we can add :)